### ISSUE TITLE
Specify a starting path from which Composer installs the dependancy

### ISSUE THREAD

**Issue by agiuliano** [composer/composer#2495]
Supposing you have a sandbox, with its structure

```
config/
src/
ext/
web/
...
```

in `src/MyLib/` there is the library I want to package with Composer.
So we have the sandbox with a specific configuration for the library MyLib and its dependancies.
I want to distribute MyLib over other projects, say proj1, proj2.

I want to version the sandbox, so I can easily clone it and work on it without warrying every time to confure properly the Lib with dependancies.
But I'd also like to add a composer.json in my sandbox root to tell that MyLib is the starting point from wich Composer have to clone the library on other `proj_i`.

In short, I want only one repository, in which there is the whole sandbox but Composer should be clever enough to clone not all the sandbox, but only the library starting from a specific point from the sandbox repository.
